libraries/libfc/include/fc/container/ordered_diff.hpp

```diff
@@ -94,16 +107,24 @@ class ordered_diff {
    template <typename X>
    requires std::same_as<std::decay_t<X>, diff_result>
    static Container<T> apply_diff(Container<T>&& container, X&& diff) {
+      FC_ASSERT(diff.remove_indexes.size() <= MAX_NUM_ARRAY_ELEMENTS);
+      FC_ASSERT(diff.insert_indexes.size() <= MAX_NUM_ARRAY_ELEMENTS);
+
       // Remove from the source based on diff.remove_indexes
       std::ptrdiff_t offset = 0;
       for (SizeType index : diff.remove_indexes) {
+         FC_ASSERT(index + offset <= container.size(), "diff.remove_indexes index ${idx} + offset ${o} not in range ${s}",
```

Comment:
should be `<` I believe.

Reply:
`.end()` i.e. `.begin() + 0` is allowed and needed.

Reply:
doc of `std::vector::erase`:

The iterator pos must be valid and dereferenceable. Thus the end() iterator (which is valid, but is not dereferenceable) cannot be used as a value for pos.

Reply:
Oh, right, I was thinking of insert.